near-sdk/src/lib.rs

```diff
@@ -277,6 +277,63 @@ extern crate quickcheck;
 /// }
 /// ```
 ///
+/// ## `#[callback_unwrap]` (annotates function parameters)
+///
+/// Automatically unwraps the successful result of a callback from a cross-contract call.
+/// Used on parameters in callback methods that are invoked as part of a cross-contract call chain.
+/// If the promise fails, the method will panic with the error message.
+///
+/// This attribute is commonly used with [`PromiseOrValue<T>`], which represents either an immediate value
+/// or a promise that will resolve to that value.
+///
+/// ### Example with Cross-Contract Factorial:
+///
+/// ```rust
+/// # use near_sdk::{near, env, PromiseOrValue};
+/// #[near(contract_state)]
+/// #[derive(Default)]
+/// pub struct CrossContract {}
+///
+/// #[near]
+/// impl CrossContract {
+///     pub fn factorial(&self, n: u32) -> PromiseOrValue<u32> {
+///         if n <= 1 {
+///             return PromiseOrValue::Value(1);
+///         }
+///         let account_id = env::current_account_id();
+///         Self::ext(account_id.clone())
+///             .factorial(n - 1)
+///             .then(Self::ext(account_id).factorial_mult(n))
+///             .into()
+///     }
+///
+///     #[private]
+///     pub fn factorial_mult(&self, n: u32, #[callback_unwrap] cur: u32) -> u32 {
+///         n * cur
+///     }
+/// }
+/// ```
+///
+/// ### Error Handling Alternative
+///
+/// If you need to handle failed promises explicitly, receive the `Result<T, PromiseError>` directly
+/// instead of using `#[callback_unwrap]`:
+///
+/// ```rust
+/// # use near_sdk::{near, PromiseError};
+/// # #[near(contract_state)]
+/// # pub struct Contract {}
+/// # #[near]
+/// # impl Contract {
+///     #[private] 
+///     pub fn handle_callback(&mut self, result: Result<u8, PromiseError>) {
+///         match result {
+///             Ok(value) => { /* Handle success */ },
+///             Err(err) => { /* Handle error */ },
+///         }
+///     }
+/// # }
+/// ```
```

Comment:
this example looks a superficial copy-paste from elsewhere, and it's not very clear how it's an alternative one to the first heading `### Example with Cross-Contract Factorial`, or what else it is an alternative to. 

if it's indeed an alternative to the first heading, it should accomplish the same purpose as first heading, but without using the `#[callback_unwrap]`, thus illustrating the difference

Reply:
@TobieTom i suggest that you remove this example completely, as from what i can see the `Result<u8, PromiseError>` is only supposed to be used together with another attribute `#[callback_result]` 

```rust
        #[callback_result] c: Result<u8, PromiseError>,
```